app/Booking.php

```diff
@@ -1,61 +0,0 @@
-<?php
-
-namespace App;
-
-use App\Exceptions\OverlappingBookingException;
-use Carbon\Carbon;
-use Illuminate\Database\Eloquent\Model;
-use Illuminate\Database\Eloquent\Relations\BelongsTo;
-
-class Booking extends Model
-{
-    protected $fillable = ['user_id', 'position_id', 'from', 'to'];
-    protected $dates = ['from', 'to'];
-
-    public function position(): BelongsTo
-    {
-        return $this->belongsTo(Position::class);
-    }
-
-    public static function canBeMade($position_id, Carbon $from, Carbon $to, $excludeID = null)
-    {
-        $query = self::where('position_id', $position_id);
-
-        if ($excludeID) {
-            $query->where('id', '!=', $excludeID);
-        }
-
-        // Find between the times being booked for
-        $query->where(function ($query) use ($from, $to) {
-            // Where the start date is inside the booked time
-            $query->where(function ($query) use ($from, $to) {
-                $query->where('from', '>', $from)
-                        ->where('from', '<', $to);
-                // Or where the end date is inside the booked time
-            })->orWhere(function ($query) use ($from, $to) {
-                $query->where('to', '>', $from)
-                        ->where('to', '<', $to);
-                // Or where the times are the same
-            })->orWhere(['from' => $from, 'to' => $to]);
-        });
-
-        return $query->doesntExist();
-    }
-
-    public static function boot()
-    {
-        parent::boot();
-
-        self::creating(function (self $booking) {
-            if (! self::canBeMade($booking->position_id, $booking->from, $booking->to)) {
```

Comment:
I know we disagree on this, but I still think this is the best catch-all method for validating a booking. Would mean don't need custom field resolver for create or update.  @AndyTWF opinions?

Reply:
We could provide a directive to simplify the process (I have an idea on this, but requires some leg work). My opinion is that as we only provide one method of manipulating the DB (via the GraphQL API layer), this is the best way to validate what get puts in. I see your point, but using its self to validate itself before it's created means that it's much harder to test and change.

Reply:
I would disagree that its using itself to validate itself - I see it as just moving where we do the check, i.e.. your method you do it once the api request comes in, mine does it when creating/updating. But if we end up wanting to, for example, perform some logic (say a mentor changes a mentoring session for EGKK_TWR to EGLL_TWR), we would have implement the validity check again in that resolver to check that the new booking doesn't clash, before changing it back end.